src/iuse.cpp

```diff
@@ -4667,17 +4667,47 @@ int iuse::oxytorch( player *p, item *it, bool, const tripoint & )
         return 0;
     }
 
-    const cata::optional<tripoint> pnt_ = choose_adjacent( _( "Cut up metal where?" ) );
-    if( !pnt_ ) {
-        return 0;
-    }
-    const tripoint pnt = *pnt_;
+    const static std::set<ter_id> allowed_ter_id {
+        t_chainfence_posts,
+        t_window_enhanced,
+        t_window_enhanced_noglass,
+        t_chainfence,
+        t_chaingate_c,
+        t_chaingate_l,
+        t_bars,
+        t_window_bars_alarm,
+        t_window_bars,
+        t_reb_cage,
+        t_door_metal_locked,
+        t_door_metal_c,
+        t_door_bar_c,
+        t_door_bar_locked,
+        t_door_metal_pickable
+    };
+    const static std::set<furn_id> allowed_furn_id {
+        f_rack
+    };
 
-    if( pnt == p->pos() ) {
-        add_msg( m_info, _( "Yuck.  Acetylene gas smells weird." ) );
+    const std::function<bool( tripoint )> f = []( tripoint p ) {
+        if( p == g->u.pos() ) {
+            return false;
+        }
+        const ter_id ter = g->m.ter( p );
+        const auto furn = g->m.furn( p );
+
+        const bool is_allowed = ( allowed_ter_id.find( ter ) != allowed_ter_id.end() ) ||
+                                ( allowed_furn_id.find( furn ) != allowed_furn_id.end() ) ;
+        return is_allowed;
+    };
+
+    const cata::optional<tripoint> pnt_ = choose_adjacent_highlight(
+            _( "Cut up metal where?" ), f, false, true );
+    if( !pnt_ ) {
+        p->add_msg_if_player( m_info, _( "You can't cut that." ) );
```

Comment:
When the player escapes from the query (triggers the "QUIT" action), it will show a message from within `chose_adjacent`, and it will show this messages as well (Wait, I did not select a direction to cut at all, how come the game knows I can not cut that - whatever "that" is?)

This should probably just return from the function without printing any further message.

Same for similar places above and below.

Reply:
Thought about it. Tried both ways. 
Yes, you are probably right. Will fix it.